src/api/app/validators/workflow_event_filter_validator.rb

```diff
@@ -0,0 +1,16 @@
+class WorkflowEventFilterValidator < ActiveModel::Validator
+  SUPPORTED_FILTER_EVENTS = ['push', 'pull_request'].freeze
+
+  def validate(workflow)
+    # FIXME: replace private method calls
+    if workflow.send(:supported_filters).key?(:event)
+      # The filters method is needed for the branch_matches_branches_filter that will eventually be a validator.
+      workflow.errors.add(:base, "Workflow filter not supported: #{workflow.filters[:event]}") unless
+        SUPPORTED_FILTER_EVENTS.include?(workflow.filters[:event])
+    else
+      workflow.errors.add(:base, 'Workflow filter not present')
+    end
+
+    false
```

Comment:
Returning a boolean isn't needed for a custom validator. This line can be removed.